### PR TITLE
Add initial version of node-buffer-blob

### PR DESCRIPTION
**Pull request by sigma-andex**
First version of node-buffer-blob

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      - name: Verify Bower & Pulp
+        run: |
+          npm install bower pulp@16.0.0-0
+          npx bower install
+          npx pulp build -- --censor-lib --strict
+          if [ -d "test" ]; then
+            npx pulp test
+          fi
+          name: CI
```
Comment (JordanMartinez) [inline .github/workflows/ci.yml]: Is this the cause of the CI issue? Seems like this line should be deleted.

Comment (sigma-andex) [reply .github/workflows/ci.yml]: Geez, thanks! I'm blind!

**Comment by sigma-andex** [comment]
The node constructor uses a union type, accepting a list of blobs, a list of arraybuffers, a list of strings, ....
I have created constructor functions `fromArrays`, `fromBlobs` for the moment. Not sure if there is a better way to do it without depending on something like `untagged-union`..

**Comment by sigma-andex** [comment]
Apart from this, I think this initial pr is ready for review.

**Comment by sigma-andex** [comment]
🏓 @JordanMartinez

File: package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "rimraf": "^3.0.2"
+  },
+  "scripts": {
+    "clean": "rimraf output && rimraf .pulp-cache",
+    "build": "eslint src && pulp build -- --censor-lib --strict",
+    "test": "pulp test -- --censor-lib --strict"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/purescript-node/purescript-node-buffer-blob.git"
```
Comment (JordanMartinez) [inline package.json]: Could you drop the `git+` part here?

File: src/Node/Buffer/Blob.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  return function () {
+    return blob.stream();
+  };
+};
+
+export const text = function(blob) {
+  return function() {
+    return blob.text();
+  };
+};
```
Comment (JordanMartinez) [inline src/Node/Buffer/Blob.js:68]: There's no `blob.type` FFI?

File: test/Node/Buffer/Blob.purs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      blob = Blob.fromString expected Nothing
+    actual <- Promise.toAffE $ Blob.text blob
+    liftEffect $ assertEqual { actual, expected }
+
+  testFromStringWithOptions = do
+    let
+      expected = "{\"hello\":\"world\"}"
+      blob = Blob.fromString expected (Just { "type": MediaTypes.applicationJSON, endings: Blob.Transparent })
+    actual <- Promise.toAffE $ Blob.text blob
+    liftEffect $ assertEqual { actual, expected }
```
Comment (JordanMartinez) [inline test/Node/Buffer/Blob.purs]: Shouldn't there be newlines in the expected string? Otherwise, the `Blob.Transparent` part isn't being tested, right?

**Review by JordanMartinez — CHANGES_REQUESTED** [review]
Overall looks good. But there are still a few changes to be made.

Could you clarify why you didn't do FFI for `blob.type`? Or implement `Blob.from` for the following?
- Array TypedArray
- Array DataView
- Array Blob

**Comment by JordanMartinez** [comment]
Also, I wonder if the singular versions of `Blob.from` (e.g. `fromString`) should be dropped. It's not too hard to go from `fromString "foo"` to `fromStrings <<< Array.singleton` or just `fromStrings ["foo"]`. Then there's less FFI to write.

But if calling `blob.from([])` is problematic, then perhaps it should be `NonEmptyArray a` where `a` is one of those 5 types that `Blob.from` accepts.

File: src/Node/Buffer/Blob.js (hunk 3 above)
Comment (sigma-andex) [reply src/Node/Buffer/Blob.js:68]: Fixed

File: test/Node/Buffer/Blob.purs (hunk 4 above)
Comment (sigma-andex) [reply test/Node/Buffer/Blob.purs]: Added newlines.  Not sure though how this option should/could be tested

**Comment by sigma-andex** [comment]
> Also, I wonder if the singular versions of `Blob.from` (e.g. `fromString`) should be dropped. It's not too hard to go from `fromString "foo"` to `fromStrings <<< Array.singleton` or just `fromStrings ["foo"]`. Then there's less FFI to write.
> 
> But if calling `blob.from([])` is problematic, then perhaps it should be `NonEmptyArray a` where `a` is one of those 5 types that `Blob.from` accepts.

Yeah that makes sense. I changed everything to NonEmptyArrays.

**Comment by JordanMartinez** [comment]
Could you clarify why you didn't implement `Blob.from` for the following types?

    NonEmptyArray TypedArray
    NonEmptyArray DataView
    NonEmptyArray Blob

**Comment by sigma-andex** [comment]
> 



> Could you clarify why you didn't implement `Blob.from` for the following types?

Oh I just missed these, will add them

**Comment by sigma-andex** [comment]
Ok I added `fromBlobs` and `fromDataView`.  I haven't yet figured out how the TypedArray works and at this point I would prefer to have that addressed in a subsequent PR if someone requires it.

**Comment by JordanMartinez** [comment]
> Ok I added `fromBlobs` and `fromDataView`. I haven't yet figured out how the TypedArray works and at this point I would prefer to have that addressed in a subsequent PR if someone requires it.

I'm good with that.

**Pull request merged**